### PR TITLE
READY: Nodes change size based on total traffic in visible nodes

### PR DESCRIPTION
**Pull request by ghabbenjansen**
Implements issue #188

### CODE REVIEW

File: TriblerGUI/widgets/trustpage/js/data_processor.js [hunk 1]
```diff
@@ -312,6 +345,11 @@ function groupBy(list, key) {
  * @property {number} max_page_rank    - the highest page rank score in the set of nodes
  * @property {number} min_transmission - the smallest transmission (up+down) from the focus node
  * @property {number} max_transmission - the largest transmission (up+down) from the focus node
+<<<<<<< HEAD
```
Comment (rkeulemans) [inline TriblerGUI/widgets/trustpage/js/data_processor.js]: Remove this please.

File: TriblerGUI/widgets/trustpage/js/drawing.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ function selectNodes(svg) {
  * @returns D3 Selection of <svg.node> elements
  */
 function drawNodes(svg, data, on_click) {
+    console.log(data);
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/js/drawing.js]: Remove debug functionality.

File: TriblerGUI/widgets/trustpage/style.css [hunk 3]
```diff
@@ -77,3 +77,66 @@ svg {
 	0% { opacity:0; }
 	100% { opacity:100%; }
 }
+
```
Comment (MaxVanDeursen) [inline TriblerGUI/widgets/trustpage/style.css]: Why is this file changed in this PR?

File: TriblerGUI/widgets/trustpage/js/data_processor.js [hunk 4]
```diff
@@ -173,6 +178,34 @@ function addMinMaxTransmission(response, interim) {
 }
 
 /**
+ * Calculate the base and the slope for the linear function describing the amount of traffic.
+ *
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/data_processor.js]: What do you mean by 'the linear funtion describing the amount of traffic'?

File: TriblerGUI/widgets/trustpage/js/data_processor.js [hunk 5]
```diff
@@ -173,6 +178,34 @@ function addMinMaxTransmission(response, interim) {
...
+    interim.nodes.forEach(function (node) {
+       var sum = node.total_up + node.total_down;
+       if (sum < min) min = sum;
+       if (sum > max) max = sum;
+    });
+
+    min = min === Number.MAX_VALUE ? 0 : min;
+    max = max === Number.MIN_VALUE ? 1 : max;
+
+    var slope = (config.node.circle.maxRadius - config.node.circle.minRadius) / (max - min);
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/data_processor.js]: please keep config values out of this file. This can be done by moving the config part to drawing.js. This way we separate drawing logic from data logic.

File: TriblerGUI/widgets/trustpage/js/drawing.js [hunk 6]
```diff
@@ -47,21 +48,31 @@ function drawNodes(svg, data, on_click) {
         .attr("fill", function (d) {
             return getNodeColor(d, data)
         })
-        .attr("r", "0")
+        .attr("r", function (d) {
+            return getNodeRadius(d, data)
+        })
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/drawing.js]: Please remove this piece, the new nodes must start at radius 0. On line 74 you set their final value after a transition of 1000 ms.

File: TriblerGUI/widgets/trustpage/js/drawing.js [hunk 7]
```diff
@@ -79,6 +90,11 @@ function drawNodes(svg, data, on_click) {
     return groups;
 }
 
+function getNodeRadius(node, data) {
+    var nodeTraffic = node.total_up + node.total_down;
+    return data.traffic_base + (nodeTraffic * data.traffic_slope);
+}
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/drawing.js]: Add the 'config logic' from the data processor here.

File: TriblerGUI/test/widgets/trustpage/js/test_data_processor.js [hunk 8]
```diff
@@ -2,6 +2,7 @@
  * This file tests the data_processor.js file with unit tests
  */
 assert = require('assert');
+config = require("TriblerGUI/widgets/trustpage/js/style_config.js");
```
Comment (TimSpeelman) [inline TriblerGUI/test/widgets/trustpage/js/test_data_processor.js]: This line can be removed once you separate the config from the data processor.

File: TriblerGUI/test/widgets/trustpage/js/test_data_processor.js [hunk 9]
```diff
@@ -188,6 +189,44 @@ describe('data_processor.js', function () {
...
+
+          var result = processor.convertResponse(response, [
+              processor.mapNodes,
+              processor.mapEdges,
+              processor.addTrafficFunction
+          ]);
+
+          assert.equal(result.traffic_slope, config.node.circle.maxRadius - config.node.circle.minRadius);
+          assert.equal(result.traffic_base, config.node.circle.minRadius);
+       });
```
Comment (TimSpeelman) [inline TriblerGUI/test/widgets/trustpage/js/test_data_processor.js]: These lines can be removed once you separate the config from the data processor.

File: TriblerGUI/widgets/trustpage/js/data_processor.js [hunk 10]
```diff
@@ -171,6 +172,34 @@ function addMinMaxTransmission(response, interim) {
...
 /**
+ * Calculate the parameters for the linear line describing the node size based on the amount of total traffic.
+ *
+ * @param {GraphResponseData} response - The response
+ * @param {Object} interim - The interim result
+ * @returns {{traffic_min: number, traffic_slope: number}}
+ */
+function addTrafficFunction(response, interim) {
+    var min = Number.MAX_VALUE,
+        max = Number.MIN_VALUE;
```
Comment (TimSpeelman) [inline TriblerGUI/widgets/trustpage/js/data_processor.js]: Can also just be zero
